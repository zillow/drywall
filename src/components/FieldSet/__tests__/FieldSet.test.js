import React from 'react';
import TestRenderer from 'react-test-renderer';
import { FieldSet } from '../../../index';

describe('<FieldSet>', () => {
    it('renders a default field set', () => {
        const testRenderer = TestRenderer.create(
            <FieldSet legend={<legend>test legend</legend>}>form controls</FieldSet>
        );
        const tree = testRenderer.toJSON();
        expect(tree.type).toBe('fieldset');
        const legend = tree.children[0];
        expect(legend.type).toBe('legend');
        expect(legend.children).toStrictEqual(['test legend']);
        expect(tree.children[1]).toBe('form controls');
        expect(tree).toMatchSnapshot();
    });

    it('forwards ref', () => {
        const ref = React.createRef();
        TestRenderer.create(
            <FieldSet legend={<legend>test legend</legend>} ref={ref}>
                form controls
            </FieldSet>,
            {
                createNodeMock: () => 'node-mock',
            }
        );
        expect(ref.current).toBe('node-mock');
    });
});
