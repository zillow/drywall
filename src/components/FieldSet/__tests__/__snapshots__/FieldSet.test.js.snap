// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldSet> renders a default field set 1`] = `
<fieldset>
  <legend>
    test legend
  </legend>
  form controls
</fieldset>
`;
